### ISSUE TITLE
test 3 fails consistently

### ISSUE THREAD

**Issue by richardlau** [nodejs/nodereport#12]
Node v6.9.1, x64 Linux RHEL 7.2.

Had to patch the nodereport source with nodejs/nodereport#9 to be able to compile. I also needed to checkout nodereport from github as the tests are no longer pulled down when I `npm install` (I think as a result of nodejs/nodereport#8).

```
-bash-4.2$ npm test -- --save_log

> nodereport@1.0.4 test /home/users/riclau/sandbox/scratch/test/pkgtest/node_modules/nodereport
> node test/autorun.js "--save_log"

autorun.js: running child process #1 to produce NodeReport on API call
autorun.js: running child process #2 to produce NodeReport on exception
autorun.js: running child process #3 to produce NodeReport on fatal error
autorun.js: running child process #4 to produce NodeReport on SIGUSR2
autorun.js: located NodeReport:  NodeReport.20161028.143703.61091.001.txt
autorun.js: located NodeReport:  NodeReport.20161028.143703.61097.001.txt
autorun.js: located NodeReport:  NodeReport.20161028.143704.61099.001.txt
autorun.js: located NodeReport:  NodeReport.20161028.143704.61104.001.txt
autorun.js: test results:
        test 1 :  pass
        test 2 :  pass
        test 3 :  fail
        test 4 :  pass
-bash-4.2$
```

[nodereport_test.log](https://gist.github.com/richardlau/28214a917a1db97f5ef2a1f65e01c69d)

**Comment by richardlau** [comment]
I modified `test/autorun.js` so that it doesn't delete the report for test 3.
```
-bash-4.2$ npm test -- --save_log

> nodereport@1.0.4 test /home/users/riclau/sandbox/scratch/test/pkgtest/node_modules/nodereport
> node test/autorun.js "--save_log"

autorun.js: running child process #1 to produce NodeReport on API call
autorun.js: running child process #2 to produce NodeReport on exception
autorun.js: running child process #3 to produce NodeReport on fatal error
autorun.js: running child process #4 to produce NodeReport on SIGUSR2
autorun.js: located NodeReport:  NodeReport.20161028.145340.75220.001.txt
autorun.js: located NodeReport:  NodeReport.20161028.145341.75227.001.txt
autorun.js: located NodeReport:  NodeReport.20161028.145340.75226.001.txt
autorun.js: located NodeReport:  NodeReport.20161028.145341.75233.001.txt
autorun.js: test results:
        test 1 :  pass
        test 2 :  pass
        test 3 :  fail
        test 4 :  pass
-bash-4.2$
```

[nodereport_test.log and NodeReport.20161028.145341.75227.001.txt](https://gist.github.com/richardlau/149e29d4b45223a4990ab106f9163ffb)

**Comment by richardlau** [comment]
I'm assuming for now this is different to #7 as three of the tests are passing for me.

**Comment by rnchamberlain** [comment]
Test 3 is the fatal error/out of memory test. The NodeReport above is truncated after the header section, so it failed while collecting the JS stack trace. This may be another case where we can't collect the JS stack if we are actually inside GC when it failed. See node_report.cc line 439, also https://groups.google.com/forum/#!topic/v8-dev/c6vGpNMScKU

**Comment by rnchamberlain** [comment]
Reproduced this. On v6.9.1 we get a second OOM in v8 if we call the stacktrace API in this scenario. It worked OK on v6.3.0. Fix will be to remove that call for now. The v8 stderr output does show a single JS frame on OOM, so there may be something we can get for nodereport via another route at some point.

@richardlau could you test with this branch please (it is level with master, so includes https://github.com/nodejs/nodereport/pull/9 and https://github.com/nodejs/nodereport/pull/11)
`git clone -b avoid-jsstacktrace-in-gc https://github.com/rnchamberlain/nodereport`

**Comment by richardlau** [comment]
Looks good :smile: 
```
-bash-4.2$ npm test

> nodereport@1.0.4 test /home/users/riclau/sandbox/scratch/test/pkgtest/node_modules/nodereport
> node test/autorun.js

autorun.js: running child process #1 to produce NodeReport on API call
autorun.js: running child process #2 to produce NodeReport on exception
autorun.js: running child process #3 to produce NodeReport on fatal error
autorun.js: running child process #4 to produce NodeReport on SIGUSR2
autorun.js: located NodeReport:  NodeReport.20161102.134843.100555.001.txt
autorun.js: located NodeReport:  NodeReport.20161102.134844.100567.001.txt
autorun.js: located NodeReport:  NodeReport.20161102.134843.100561.001.txt
autorun.js: located NodeReport:  NodeReport.20161102.134844.100568.001.txt
autorun.js: test results:
   test 1 :  pass
   test 2 :  pass
   test 3 :  pass
   test 4 :  pass
-bash-4.2$
```

**Comment by rnchamberlain** [comment]
published in npm - nodereport 1.0.5

**Comment by richardlau** [comment]
Fixed by #16.
